### PR TITLE
[MIG] delivery_carrier_label_postlogistics: Migration to 12.0

### PR DESCRIPTION
**Pull request by sebalix**
TODO:
 - [x] Use new REST API (addressed by @jcoux)

### CODE REVIEW

File: delivery_carrier_label_postlogistics/tests/test_postlogistics.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+from odoo.tests import common
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class FakeWS(PostlogisticsWebService):
+    def generate_label(self, picking, packages, user_lang=None):
+        # call super to pass in the methods so we can check their
+        # behavior.
+        super(FakeWS, self).generate_label(picking, packages,
```
Comment (i-vyshnevska) [inline delivery_carrier_label_postlogistics/tests/test_postlogistics.py]:
```suggestion
        super().generate_label(picking, packages,
```

File: delivery_carrier_label_postlogistics/tests/test_postlogistics.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+             '.postlogistics.web_service.HttpAuthenticated')
+output_path = ('odoo.addons.delivery_carrier_label_postlogistics'
+               '.postlogistics.web_service.PostlogisticsWebService'
+               '._get_output_format')
+
+
+class TestPostlogistics(common.TransactionCase):
+
+    def setUp(self):
+        super(TestPostlogistics, self).setUp()
```
Comment (i-vyshnevska) [inline delivery_carrier_label_postlogistics/tests/test_postlogistics.py]:
```suggestion
        super().setUp()
```

File: delivery_carrier_label_postlogistics/tests/test_delivery.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    'output_format',
+    'resolution',
+]
+
+
+class TestDelivery(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestDelivery, cls).setUpClass()
```
Comment (i-vyshnevska) [inline delivery_carrier_label_postlogistics/tests/test_delivery.py]:
```suggestion
        super().setUpClass()
```

**Review by i-vyshnevska — APPROVED** [review]
LG, just code review

**Comment by sebalix** [comment]
`default_` configuration fields have a special meaning for Odoo configuration settings panel (to define default values on a given data model).
As such the following fields on 'res.company' have been renamed on purpose:

- postlogistics_default_label_layout -> postlogistics_label_layout
- postlogistics_default_output_format -> postlogistics_output_format
- postlogistics_default_resolution -> postlogistics_resolution

**Comment by rousseldenis** [comment]
@sebalix Can you fix runbot warning ?

2019-11-20 15:49:14,423 160 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (postlogistics_basic_service_ids, postlogistics_service_group_id) of delivery.carrier() have the same label: PostLogistics Service Group.

**Comment by yvaucher** [comment]
swisspost now has a REST API, https://developer.post.ch/en/digital-commerce-api

We will move forward to the new API in the current PR to implement it and drop the support for the SOAP API. Thus, consider this as WIP.

**Review by yvaucher — APPROVED** [review]
Tested locally and could print a label.

**Comment by yvaucher** [comment]
A couple of lint to fix:

```
delivery_carrier_label_postlogistics/postlogistics/web_service.py:174:20: E131 continuation line unaligned for hanging indent
delivery_carrier_label_postlogistics/postlogistics/web_service.py:450:89: E501 line too long (90 > 88 characters)
```

Plus some errors in the tests:

```
odoo.addons.delivery_carrier_label_postlogistics.tests.test_postlogistics: ` AttributeError: <module 'odoo.addons.delivery_carrier_label_postlogistics.postlogistics.web_service' from '/home/travis/build/OCA/delivery-carrier/delivery_carrier_label_postlogistics/postlogistics/web_service.py'> does not have the attribute 'Client'
```

**Comment by yvaucher** [comment]
Rebased and squashed

**Comment by sbidoul** [comment]
@yvaucher I was wondering why the README issue that Sentry reported had not been caught by the bot.
But actually you did not leave it time to finish, it's still waiting for travis on the merge branch: https://github.com/OCA/delivery-carrier/commits/12.0-ocabot-merge-pr-235-by-yvaucher-bump-nobump.

**Comment by yvaucher** [comment]
@sbidoul it seemed stuck to me as it took more than 1h and travis was already green. Seems resolved though, did you launched any action to wake up the bot?

**Comment by sbidoul** [comment]
@yvaucher no I did not do any manual action. Actually when you see travis green on the commit mentioned in the PR by the bot, it can be misleading because .pot files may have been regenerated, triggering new checks. So in doubdt, check the head of the `12.0-ocabot-merge-pr...` branch to see if there are ongoing travis checks on it.

**Pull request merged**